app/Console/ReindexOffersWithPopularityScore.php

```diff
@@ -0,0 +1,146 @@
+
+use Broadway\Domain\DomainEventStream;
+use Broadway\EventHandling\EventBusInterface;
+use CultuurNet\Broadway\EventHandling\ReplayModeEventBusInterface;
+use CultuurNet\UDB3\ReadModel\DocumentEventFactory;
+use Doctrine\DBAL\Connection;
+use Knp\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+
+class ReindexOffersWithPopularityScore extends Command
+{
+    private $allowedTypes = [
+        'event',
+        'place',
+    ];
+
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * @var EventBusInterface
+     */
+    private $eventBus;
+
+    /**
+     * @var DocumentEventFactory
+     */
+    private $eventFactoryForEvents;
+
+    /**
+     * @var DocumentEventFactory
+     */
+    private $eventFactoryForPlaces;
+
+    public function __construct(
+        Connection $connection,
+        EventBusInterface $eventBus,
+        DocumentEventFactory $eventFactoryForEvents,
+        DocumentEventFactory  $eventFactoryForPlaces
+    ) {
+        parent::__construct();
+        $this->connection = $connection;
+        $this->eventBus = $eventBus;
+        $this->eventFactoryForEvents = $eventFactoryForEvents;
+        $this->eventFactoryForPlaces = $eventFactoryForPlaces;
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->setName('offer:reindex-offers-with-popularity')
+            ->setDescription('Reindex events or places that have a popularity score.')
+            ->addArgument(
+                'type',
+                InputArgument::REQUIRED,
+                'The type of the offer, either place or event.'
+            )
+            ->addOption(
+                'force',
+                null,
+                InputOption::VALUE_NONE,
+                'Skip confirmation.'
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        if (!$this->eventBus instanceof ReplayModeEventBusInterface) {
+            throw new \InvalidArgumentException('The event bus should implement the "ReplayModeEventBusInterface".');
+        }
+
+        $type = $input->getArgument('type');
+        if (!\in_array($type, $this->allowedTypes, true)) {
+            throw new \InvalidArgumentException('The type "' . $type . '" is not support. Use event or place.');
+        }
+
+        $offerIds = $this->getOfferIds($type);
+        if (count($offerIds) < 1) {
+            $output->writeln('No ' . $type . 's found with a popularity.');
+            return 0;
+        }
+
+        if (!$this->askConfirmation($input, $output, $type, count($offerIds))) {
+            return 0;
+        }
+        if ($this->eventBus instanceof ReplayModeEventBusInterface) {
```

Comment:
This `if` check can be removed since we throw an exception higher up if the event bus is not an instance of `ReplayModeEventBusInterface` (regardless of my suggestion to move it as a type check to the constructor)

Reply:
No longer needed after injecting `ReplayModeEventBusInterface`.